File: src/core/constants/adaptiveThresholds.ts
/**
 * BilimYo‘l Adaptive Thresholds Configuration
 * 
 * Rules:
 * score < 50     -> reinforcement (requires remediation node)
 * 50 <= score < 70 -> targeted_practice
 * 70 <= score < 85 -> standard
 * score >= 85    -> advanced
 */
export const ADAPTIVE_THRESHOLDS = {
  REINFORCEMENT_MAX: 49,
  TARGETED_PRACTICE_MIN: 50,
  TARGETED_PRACTICE_MAX: 69,
  STANDARD_MIN: 70,
  STANDARD_MAX: 84,
  ADVANCED_MIN: 85,
} as const;

export type RouteStrategy = 'reinforcement' | 'targeted_practice' | 'standard' | 'advanced';

export function getRouteStrategy(score: number): RouteStrategy {
  if (score <= ADAPTIVE_THRESHOLDS.REINFORCEMENT_MAX) {
    return 'reinforcement';
  }
  if (score <= ADAPTIVE_THRESHOLDS.TARGETED_PRACTICE_MAX) {
    return 'targeted_practice';
  }
  if (score <= ADAPTIVE_THRESHOLDS.STANDARD_MAX) {
    return 'standard';
  }
  return 'advanced';
}

/**
 * Improvement delta constants
 */
export const REINFORCEMENT_SCORE_BUMP = 22; // e.g. 41% -> 63% or 43% -> 65%
export const REINFORCEMENT_XP_REWARD = 30; // +30 XP
export const LESSON_XP_REWARD = 20;

File: mobile/lib/core/constants/adaptive_thresholds.dart
/// BilimYo‘l Adaptive Thresholds Configuration
///
/// Rules:
/// score < 50        -> reinforcement (requires remediation node)
/// 50 <= score < 70  -> targeted_practice
/// 70 <= score < 85  -> standard
/// score >= 85       -> advanced
class AdaptiveThresholds {
  static const int reinforcementMax = 49;
  static const int targetedPracticeMin = 50;
  static const int targetedPracticeMax = 69;
  static const int standardMin = 70;
  static const int standardMax = 84;
  static const int advancedMin = 85;

  static const int reinforcementScoreBump = 22; // e.g. 41% -> 63% or 43% -> 65%
  static const int reinforcementXpReward = 30; // +30 XP
  static const int lessonXpReward = 20; // +20 XP
}

enum RouteStrategy {
  reinforcement,
  targetedPractice,
  standard,
  advanced,
}

RouteStrategy getRouteStrategy(int score) {
  if (score <= AdaptiveThresholds.reinforcementMax) {
    return RouteStrategy.reinforcement;
  }
  if (score <= AdaptiveThresholds.targetedPracticeMax) {
    return RouteStrategy.targetedPractice;
  }
  if (score <= AdaptiveThresholds.standardMax) {
    return RouteStrategy.standard;
  }
  return RouteStrategy.advanced;
}
